cumulus_library/studies/core/builder_core_medication.py

```diff
@@ -18,17 +18,50 @@ def _check_data_in_fields(self, cursor, schema: str):
         """Validates whether either observed medication source is present"""
 
         data_types = {
+            "inline": False,
             "by_contained_ref": False,
             "by_external_ref": False,
         }
 
+        table = "medicationrequest"
+        base_col = "medicationcodeableconcept"
+
         with get_progress_bar(transient=True) as progress:
             task = progress.add_task(
                 "Detecting available medication sources...",
-                total=5,
+                total=8,
             )
 
-            # Validating medication requests
+            # inline medications from FHIR medication
+            query = get_is_table_not_empty_query(table, base_col)
+            cursor.execute(query)
+            progress.advance(task)
+            if cursor.fetchone() is not None:
+                query = get_column_datatype_query(schema, table, base_col)
+                cursor.execute(query)
+                progress.advance(task)
+                if "coding" in str(cursor.fetchone()[0]):
+                    query = get_is_table_not_empty_query(
+                        table,
+                        "t1.row1",
+                        [
+                            {
+                                "source_col": f"{base_col}.coding",
+                                "table_alias": "t1",
+                                "row_alias": "row1",
+                            }
+                        ],
+                    )
+                    cursor.execute(query)
+                    progress.advance(task)
+                    if cursor.fetchone() is not None:
+                        data_types["inline"] = True
+                else:
+                    progress.advance(task)
+            else:
+                progress.advance(task, advance=2)
```

Comment:
so this is the technical debt I was discussing - this looks very similar to the code in builder_encounter_coding, but it's structurally different in one important way: it uses the table schema for determining if coding is an available field, rather than relying on table length. This is slightly safer for the case where we've got unusual 'data is not present' conditions.

My intent is to move this to some kind of template_sql/utils kind of location, and import it here and in the encounter builder - but in order to do that, i'll need to split apart the array vs non array handling, and write a similar version for array handling. I'm not doing this as part of this PR due to time reasons.